test/opt/local_ssa_elim_test.cpp

```diff
@@ -2024,6 +2024,788 @@ OpFunctionEnd
   SinglePassRunToBinary<SSARewritePass>(text, false);
 }
 
+TEST_F(LocalSSAElimTest, DebugForLoop) {
+  // #version 140
+  //
+  // in vec4 BC;
+  // out float fo;
+  //
+  // void main()
+  // {
+  //     float f = 0.0;
+  //     for (int i=0; i<4; i++) {
+  //       f = f + BC[i];
+  //     }
+  //     fo = f;
+  // }
+
+  const std::string text = R"(
+; CHECK: [[f_name:%\w+]] = OpString "f"
+; CHECK: [[i_name:%\w+]] = OpString "i"
+; CHECK: [[dbg_f:%\w+]] = OpExtInst %void [[ext:%\d+]] DebugLocalVariable [[f_name]]
+; CHECK: [[dbg_i:%\w+]] = OpExtInst %void [[ext]] DebugLocalVariable [[i_name]]
+
+; CHECK:      OpStore %f %float_0
+; CHECK:      OpStore %i %int_0
+; CHECK:      OpExtInst %void [[ext]] DebugDeclare [[dbg_f]] %f
+; CHECK:      OpExtInst %void [[ext]] DebugDeclare [[dbg_i]] %i
+; CHECK:      [[phi0:%\w+]] = OpPhi %float %float_0
+; CHECK-NEXT: OpExtInst %void [[ext]] DebugValue [[dbg_f]] [[phi0]]
```

Comment:
Can you have debug value instruction mixed in with the OpPhi instructions?

Reply:
The spec says:

> Within a block, this instruction must appear before all non-OpPhi instructions (except for OpLine and OpNoLine, which can be mixed with OpPhi).

I do not see anything in the debug instruction spec that changes this.  Personally, it does not make sense that a debug value or debug declare should be mixed with the OpPhi instructions.  Technically the spec says they are not allowed (regardless of what spirv-val does).

Reply:
We should double check this.

Reply:
Oh I see. Now I checked it in the description of `OpPhi`. I fixed it.